File: mobile/lib/features/employees/domain/entities/employee.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../../../auth/domain/entities/role.dart';

part 'employee.freezed.dart';

/// Someone on the payroll. Mirrors
/// backend/DTOs/Employees/EmployeeResponseDto.cs — notably never a password.
@freezed
abstract class Employee with _$Employee {
  const factory Employee({
    required int id,
    required String employeeCode,
    required String firstName,
    required String lastName,
    required String email,
    required int departmentId,
    required String departmentName,
    required Role role,
    required bool isActive,
    required DateTime createdAt,
    String? phoneNumber,
    int? managerId,
    String? managerName,
  }) = _Employee;

  const Employee._();

  String get fullName => '$firstName $lastName';
}

File: frontend/src/types/employee.ts
// Mirrors backend/DTOs/Employees/*.cs
import type { Role } from "./auth";

export interface Employee {
  id: number;
  employeeCode: string;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string | null;
  departmentId: number;
  departmentName: string;
  role: Role;
  managerId: number | null;
  managerName: string | null;
  isActive: boolean;
  createdAt: string;
}

export interface EmployeeCreateRequest {
  employeeCode: string;
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  phoneNumber?: string;
  departmentId: number;
  role: Role;
  managerId?: number;
}

// No password field — matches the backend, which handles password changes
// through a separate flow rather than a routine profile edit.
export interface EmployeeUpdateRequest {
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber?: string;
  departmentId: number;
  role: Role;
  managerId?: number;
  isActive: boolean;
}
